cmd/collector/app/handler.go

```diff
@@ -80,6 +85,51 @@ func (aH *APIHandler) saveSpan(w http.ResponseWriter, r *http.Request) {
 		} else {
 			w.WriteHeader(http.StatusOK)
 		}
+
+	case "zipkin.thrift":
+		bodyBytes, err := ioutil.ReadAll(r.Body)
+		r.Body.Close()
+		if err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		buffer := thrift.NewTMemoryBuffer()
+		if _, err = buffer.Write(bodyBytes); err != nil {
+			http.Error(w, fmt.Sprintf(unableToReadBodyErrFormat, err), http.StatusBadRequest)
+			return
+		}
+
+		transport := thrift.NewTBinaryProtocolTransport(buffer)
+		_, size, err := transport.ReadListBegin()
```

Comment:
Well, then you'll need to find a way to simulate the errors in reading the list to maintain the coverage. 

We could also do a hack - pad the bytes from HTTP with a few bytes that this class needs:

https://github.com/uber/jaeger-client-go/blob/master/thrift-gen/zipkincore/zipkincollector.go#L233